File: Source/C#/DevnologyFitnesseDojo/DevnologyFitnesseDojo/Fitnesse/Amasun/Book/BuyBookScenarioForCustomer.cs
using System;
using DevnologyFitnesseDojo.Domain.Books;
using DevnologyFitnesseDojo.Domain.Customers;
using DevnologyFitnesseDojo.Domain.Orders;

namespace DevnologyFitnesseDojo.Fitnesse.Amasun.Book
{
    public class BuyBookScenarioForCustomer
    {
        private readonly BookService bookService = new BookService();
        private readonly Domain.Customers.Customer customer;
        private readonly Order order;

        public BuyBookScenarioForCustomer(string customerName)
        {
            customer = CustomerRepository.FindByName(customerName);

            order = new Order(customer);
        }

        public bool CustomerBuysBooksWithTitle(int amount, String title)
        {
            Domain.Books.Book book = Inventory.FindByTitle(title);

            order.AddBook(amount, book);

            return true;
        }

        public bool PlaceOrder()
        {
            bookService.BuyBooks(order);

            return true;
        }

        public int AppliedDiscount()
        {
            return order.Discount;
        }

        public int ChargedSubtotal()
        {
            return order.Subtotal;
        }

        public int ChargedTotal()
        {
            return order.Total;
        }
    }
}

File: Source/Java/nl/devnology/fitnesse/amasun/book/BuyBookScenarioForCustomer.java
package nl.devnology.fitnesse.amasun.book;

import nl.devnology.domain.book.Book;
import nl.devnology.domain.book.BookService;
import nl.devnology.domain.book.Inventory;
import nl.devnology.domain.customer.Customer;
import nl.devnology.domain.customer.Customers;
import nl.devnology.domain.order.Order;

/**
 * @author Erik Pragt
 */
public class BuyBookScenarioForCustomer {
    private BookService bookService = new BookService();

    private Integer beforeMoney;
    private Customer customer;
    private Order order;

    public BuyBookScenarioForCustomer(String customerName) {
        customer = Customers.findByName(customerName);


        beforeMoney = customer.getMoney();

        order = new Order(customer);
    }

    public boolean customerBuysBooksWithTitle(Integer amount, String title) {
        Book book = Inventory.get().findByTitle(title);

        order.addBook(amount, book);

        return true;
    }

    public boolean placeOrder() {
        bookService.buyBooks(order);

        return true;
    }

    public Integer customerPayed() {
        return beforeMoney - customer.getMoney();

    }
}
